packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkTest.kt

```diff
@@ -1505,6 +1506,40 @@ internal class DdSdkTest {
 
     // endregion
 
+    // region resource mapper
+
+    @Test
+    fun `𝕄 set a resource mapper that drops flagged resources 𝕎 initialize() {}`(
+        @Forgery resourceEvent: ResourceEvent,
+    ) {
+        // Given
+        val configCaptor = argumentCaptor<Configuration>()
+
+        // When
+        testedBridgeSdk.initialize(fakeConfiguration.toReadableJavaOnlyMap(), mockPromise)
+
+        // Then
+        verify(mockDatadog).initialize(
+            same(mockContext),
+            any(),
+            configCaptor.capture(),
+            any()
+        )
+        assertThat(configCaptor.firstValue)
+            .hasField("rumConfig") {
+                val resourceMapper = it
+                    .getActualValue<EventMapper<ResourceEvent>>("rumEventMapper")
+                val notDroppedEvent = resourceMapper.map(resourceEvent)
+                assertThat(notDroppedEvent).isNotNull
+
+                resourceEvent.context?.additionalProperties?.put("_dd.resource.drop_resource", true)
```

Comment:
it is better to split this test into 2 different tests: one that checks that nothing is dropped, another checks that event is dropped. This line should be in `Given`.